app/interactors/request_password_reset/generate_reset_token.rb

```diff
@@ -0,0 +1,19 @@
+class RequestPasswordReset
+  class GenerateResetToken
+    include Interactor
+
+    delegate :email, to: :context
+
+    def call
+      user.regenerate_password_reset_token
+      user.password_reset_sent_at = Time.current
+      user.save!
+    end
+
+    def user
+      context.user ||= User.find_by!(email: email)
+    rescue ActiveRecord::RecordNotFound
+      context.fail!(error: I18n.t("password_recovery.not_found"))
```

Comment:
```suggestion
      context.fail!(error_data: I18n.t("password_recovery.not_found"))
```

I think that you have not spec for this case